upstream-sha256: d0744495993ed0f158d181146be5fd332da2bb448dc91b50d8ca45f2efc5765b
actual-sha256: f035162779e8cb0198d7cbd82df7b8946140591e372da3ac002b7e79d5ea975a
--- upstream.expected
+++ tolk.actual
@@ -22,84 +22,90 @@
   21 Ops.CAST            :            dtypes.float                             ['-7.0']                         dtypes.float
   22 Ops.CMPLT           :            dtypes.bool                              [19, 21]                         None
   23 Ops.WHERE           :            dtypes.float                             [22, 21, 19]                     None
-  24 Ops.CONST           :            dtypes.weakfloat                         []                               2.4554669857025146
-  25 Ops.CAST            :            dtypes.float                             ['2.4554669857025146']           dtypes.float
-  26 Ops.MUL             :            dtypes.float                             [23, 25]                         None
-  27 Ops.EXP2            :            dtypes.float                             [26]                             None
-  28 Ops.CONST           :            dtypes.weakfloat                         []                               1.0
-  29 Ops.CAST            :            dtypes.float                             ['1.0']                          dtypes.float
-  30 Ops.ADD             :            dtypes.float                             [27, 29]                         None
-  31 Ops.RECIPROCAL      :            dtypes.float                             [30]                             None
-  32 Ops.MUL             :            dtypes.float                             [23, 31]                         None
-  33 Ops.CONST           :            dtypes.weakint                           []                               1
-  34 Ops.CAST            :            dtypes.int                               ['1']                            dtypes.int
-  35 Ops.INDEX           :            dtypes.float                             [15, 34]                         None
-  36 Ops.CMPLT           :            dtypes.bool                              [35, 21]                         None
-  37 Ops.WHERE           :            dtypes.float                             [36, 21, 35]                     None
-  38 Ops.NEG             :            dtypes.float                             [37]                             None
-  39 Ops.CMPLT           :            dtypes.bool                              [38, 21]                         None
-  40 Ops.WHERE           :            dtypes.float                             [39, 21, 38]                     None
-  41 Ops.SUB             :            dtypes.float                             [29, 40]                         None
-  42 Ops.MUL             :            dtypes.float                             [32, 41]                         None
-  43 Ops.NEG             :            dtypes.float                             [42]                             None
-  44 Ops.INDEX           :            dtypes.float                             [15, 5]                          None
-  45 Ops.NEG             :            dtypes.float                             [44]                             None
-  46 Ops.CMPLT           :            dtypes.bool                              [45, 21]                         None
-  47 Ops.WHERE           :            dtypes.float                             [46, 21, 45]                     None
-  48 Ops.MUL             :            dtypes.float                             [47, 25]                         None
-  49 Ops.EXP2            :            dtypes.float                             [48]                             None
-  50 Ops.ADD             :            dtypes.float                             [49, 29]                         None
-  51 Ops.RECIPROCAL      :            dtypes.float                             [50]                             None
-  52 Ops.MUL             :            dtypes.float                             [47, 51]                         None
-  53 Ops.INDEX           :            dtypes.float                             [15, 12]                         None
-  54 Ops.CMPLT           :            dtypes.bool                              [53, 21]                         None
-  55 Ops.WHERE           :            dtypes.float                             [54, 21, 53]                     None
-  56 Ops.NEG             :            dtypes.float                             [55]                             None
-  57 Ops.CMPLT           :            dtypes.bool                              [56, 21]                         None
-  58 Ops.WHERE           :            dtypes.float                             [57, 21, 56]                     None
-  59 Ops.SUB             :            dtypes.float                             [29, 58]                         None
-  60 Ops.MUL             :            dtypes.float                             [52, 59]                         None
+  24 Ops.NEG             :            dtypes.float                             [23]                             None
+  25 Ops.CONST           :            dtypes.weakfloat                         []                               1.7020000219345093
+  26 Ops.CAST            :            dtypes.float                             ['1.7020000219345093']           dtypes.float
+  27 Ops.MUL             :            dtypes.float                             [24, 26]                         None
+  28 Ops.CONST           :            dtypes.weakfloat                         []                               -1.4426950216293335
+  29 Ops.CAST            :            dtypes.float                             ['-1.4426950216293335']          dtypes.float
+  30 Ops.MUL             :            dtypes.float                             [27, 29]                         None
+  31 Ops.EXP2            :            dtypes.float                             [30]                             None
+  32 Ops.CONST           :            dtypes.weakfloat                         []                               1.0
+  33 Ops.CAST            :            dtypes.float                             ['1.0']                          dtypes.float
+  34 Ops.ADD             :            dtypes.float                             [31, 33]                         None
+  35 Ops.RECIPROCAL      :            dtypes.float                             [34]                             None
+  36 Ops.MUL             :            dtypes.float                             [24, 35]                         None
+  37 Ops.CONST           :            dtypes.weakint                           []                               1
+  38 Ops.CAST            :            dtypes.int                               ['1']                            dtypes.int
+  39 Ops.INDEX           :            dtypes.float                             [15, 38]                         None
+  40 Ops.CMPLT           :            dtypes.bool                              [39, 21]                         None
+  41 Ops.WHERE           :            dtypes.float                             [40, 21, 39]                     None
+  42 Ops.NEG             :            dtypes.float                             [41]                             None
+  43 Ops.CMPLT           :            dtypes.bool                              [42, 21]                         None
+  44 Ops.WHERE           :            dtypes.float                             [43, 21, 42]                     None
+  45 Ops.SUB             :            dtypes.float                             [33, 44]                         None
+  46 Ops.MUL             :            dtypes.float                             [36, 45]                         None
+  47 Ops.INDEX           :            dtypes.float                             [15, 5]                          None
+  48 Ops.NEG             :            dtypes.float                             [47]                             None
+  49 Ops.CMPLT           :            dtypes.bool                              [48, 21]                         None
+  50 Ops.WHERE           :            dtypes.float                             [49, 21, 48]                     None
+  51 Ops.NEG             :            dtypes.float                             [50]                             None
+  52 Ops.MUL             :            dtypes.float                             [51, 26]                         None
+  53 Ops.MUL             :            dtypes.float                             [52, 29]                         None
+  54 Ops.EXP2            :            dtypes.float                             [53]                             None
+  55 Ops.ADD             :            dtypes.float                             [54, 33]                         None
+  56 Ops.RECIPROCAL      :            dtypes.float                             [55]                             None
+  57 Ops.MUL             :            dtypes.float                             [51, 56]                         None
+  58 Ops.INDEX           :            dtypes.float                             [15, 12]                         None
+  59 Ops.CMPLT           :            dtypes.bool                              [58, 21]                         None
+  60 Ops.WHERE           :            dtypes.float                             [59, 21, 58]                     None
   61 Ops.NEG             :            dtypes.float                             [60]                             None
-  62 Ops.ADD             :            dtypes.int                               [13, 8]                          None
-  63 Ops.SHRINK          :            dtypes.float                             [10, 62, 8]                      None
-  64 Ops.LOAD            :            dtypes.float                             [63]                             None
-  65 Ops.INDEX           :            dtypes.float                             [64, 17]                         None
-  66 Ops.NEG             :            dtypes.float                             [65]                             None
-  67 Ops.CMPLT           :            dtypes.bool                              [66, 21]                         None
-  68 Ops.WHERE           :            dtypes.float                             [67, 21, 66]                     None
-  69 Ops.MUL             :            dtypes.float                             [68, 25]                         None
-  70 Ops.EXP2            :            dtypes.float                             [69]                             None
-  71 Ops.ADD             :            dtypes.float                             [70, 29]                         None
-  72 Ops.RECIPROCAL      :            dtypes.float                             [71]                             None
-  73 Ops.MUL             :            dtypes.float                             [68, 72]                         None
-  74 Ops.INDEX           :            dtypes.float                             [64, 34]                         None
-  75 Ops.CMPLT           :            dtypes.bool                              [74, 21]                         None
-  76 Ops.WHERE           :            dtypes.float                             [75, 21, 74]                     None
-  77 Ops.NEG             :            dtypes.float                             [76]                             None
-  78 Ops.CMPLT           :            dtypes.bool                              [77, 21]                         None
-  79 Ops.WHERE           :            dtypes.float                             [78, 21, 77]                     None
-  80 Ops.SUB             :            dtypes.float                             [29, 79]                         None
-  81 Ops.MUL             :            dtypes.float                             [73, 80]                         None
-  82 Ops.NEG             :            dtypes.float                             [81]                             None
-  83 Ops.INDEX           :            dtypes.float                             [64, 5]                          None
-  84 Ops.NEG             :            dtypes.float                             [83]                             None
-  85 Ops.CMPLT           :            dtypes.bool                              [84, 21]                         None
-  86 Ops.WHERE           :            dtypes.float                             [85, 21, 84]                     None
-  87 Ops.MUL             :            dtypes.float                             [86, 25]                         None
-  88 Ops.EXP2            :            dtypes.float                             [87]                             None
-  89 Ops.ADD             :            dtypes.float                             [88, 29]                         None
-  90 Ops.RECIPROCAL      :            dtypes.float                             [89]                             None
-  91 Ops.MUL             :            dtypes.float                             [86, 90]                         None
-  92 Ops.INDEX           :            dtypes.float                             [64, 12]                         None
-  93 Ops.CMPLT           :            dtypes.bool                              [92, 21]                         None
-  94 Ops.WHERE           :            dtypes.float                             [93, 21, 92]                     None
-  95 Ops.NEG             :            dtypes.float                             [94]                             None
-  96 Ops.CMPLT           :            dtypes.bool                              [95, 21]                         None
-  97 Ops.WHERE           :            dtypes.float                             [96, 21, 95]                     None
-  98 Ops.SUB             :            dtypes.float                             [29, 97]                         None
-  99 Ops.MUL             :            dtypes.float                             [91, 98]                         None
- 100 Ops.NEG             :            dtypes.float                             [99]                             None
- 101 Ops.STACK           :            dtypes.float                             [43, 61, 82, 100]                None
- 102 Ops.STORE           :            dtypes.void                              [9, 101]                         None
- 103 Ops.GROUP           :            dtypes.void                              [102, 102, 102, 102]             None
- 104 Ops.SINK            :            dtypes.void                              [103]                            KernelInfo(name='E_8_4', applied_opts=(Opt(op=OptOps.SPLIT, axis=0, arg=(4, AxisType.UPCAST)), Opt(op=OptOps.SPLIT, axis=0, arg=(8, AxisType.LOCAL))), opts_to_apply=None, estimates=None, beam=0)
+  62 Ops.CMPLT           :            dtypes.bool                              [61, 21]                         None
+  63 Ops.WHERE           :            dtypes.float                             [62, 21, 61]                     None
+  64 Ops.SUB             :            dtypes.float                             [33, 63]                         None
+  65 Ops.MUL             :            dtypes.float                             [57, 64]                         None
+  66 Ops.ADD             :            dtypes.int                               [13, 8]                          None
+  67 Ops.SHRINK          :            dtypes.float                             [10, 66, 8]                      None
+  68 Ops.LOAD            :            dtypes.float                             [67]                             None
+  69 Ops.INDEX           :            dtypes.float                             [68, 17]                         None
+  70 Ops.NEG             :            dtypes.float                             [69]                             None
+  71 Ops.CMPLT           :            dtypes.bool                              [70, 21]                         None
+  72 Ops.WHERE           :            dtypes.float                             [71, 21, 70]                     None
+  73 Ops.NEG             :            dtypes.float                             [72]                             None
+  74 Ops.MUL             :            dtypes.float                             [73, 26]                         None
+  75 Ops.MUL             :            dtypes.float                             [74, 29]                         None
+  76 Ops.EXP2            :            dtypes.float                             [75]                             None
+  77 Ops.ADD             :            dtypes.float                             [76, 33]                         None
+  78 Ops.RECIPROCAL      :            dtypes.float                             [77]                             None
+  79 Ops.MUL             :            dtypes.float                             [73, 78]                         None
+  80 Ops.INDEX           :            dtypes.float                             [68, 38]                         None
+  81 Ops.CMPLT           :            dtypes.bool                              [80, 21]                         None
+  82 Ops.WHERE           :            dtypes.float                             [81, 21, 80]                     None
+  83 Ops.NEG             :            dtypes.float                             [82]                             None
+  84 Ops.CMPLT           :            dtypes.bool                              [83, 21]                         None
+  85 Ops.WHERE           :            dtypes.float                             [84, 21, 83]                     None
+  86 Ops.SUB             :            dtypes.float                             [33, 85]                         None
+  87 Ops.MUL             :            dtypes.float                             [79, 86]                         None
+  88 Ops.INDEX           :            dtypes.float                             [68, 5]                          None
+  89 Ops.NEG             :            dtypes.float                             [88]                             None
+  90 Ops.CMPLT           :            dtypes.bool                              [89, 21]                         None
+  91 Ops.WHERE           :            dtypes.float                             [90, 21, 89]                     None
+  92 Ops.NEG             :            dtypes.float                             [91]                             None
+  93 Ops.MUL             :            dtypes.float                             [92, 26]                         None
+  94 Ops.MUL             :            dtypes.float                             [93, 29]                         None
+  95 Ops.EXP2            :            dtypes.float                             [94]                             None
+  96 Ops.ADD             :            dtypes.float                             [95, 33]                         None
+  97 Ops.RECIPROCAL      :            dtypes.float                             [96]                             None
+  98 Ops.MUL             :            dtypes.float                             [92, 97]                         None
+  99 Ops.INDEX           :            dtypes.float                             [68, 12]                         None
+ 100 Ops.CMPLT           :            dtypes.bool                              [99, 21]                         None
+ 101 Ops.WHERE           :            dtypes.float                             [100, 21, 99]                    None
+ 102 Ops.NEG             :            dtypes.float                             [101]                            None
+ 103 Ops.CMPLT           :            dtypes.bool                              [102, 21]                        None
+ 104 Ops.WHERE           :            dtypes.float                             [103, 21, 102]                   None
+ 105 Ops.SUB             :            dtypes.float                             [33, 104]                        None
+ 106 Ops.MUL             :            dtypes.float                             [98, 105]                        None
+ 107 Ops.STACK           :            dtypes.float                             [46, 65, 87, 106]                None
+ 108 Ops.STORE           :            dtypes.void                              [9, 107]                         None
+ 109 Ops.GROUP           :            dtypes.void                              [108, 108, 108, 108]             None
+ 110 Ops.SINK            :            dtypes.void                              [109]                            KernelInfo(name='E_8_4', applied_opts=(Opt(op=OptOps.SPLIT, axis=0, arg=(4, AxisType.UPCAST)), Opt(op=OptOps.SPLIT, axis=0, arg=(8, AxisType.LOCAL))), opts_to_apply=None, estimates=None, beam=0)
